### PR TITLE
(SERVER-989) event notifications for pool locking

### PR DESCRIPTION
**Pull request by cprice404**

### CODE REVIEW

**Comment by rlinehan** [comment]
:+1: :ship:

**Comment by cprice404** [comment]
Pushed a commit that adds better descriptions for the test behavior.

@camlow325 this isn't super-duper-duper urgent but probably will block some things for PE Ankeny at some point (e.g. I'd like to add status information about the codedir lock to the PE status endpoint, and I don't think I can do it without this PR).

File: test/integration/puppetlabs/services/jruby/jruby_locking_test.clj [hunk 1]
```diff
@@ -28,14 +47,54 @@
...
+              (is (= [:instance-requested :instance-borrowed :instance-returned] @events)))
+            (jruby-service/with-lock
+              jruby-service
+              :with-lock-events-test
+              (testing "acquiring a lock triggers 'lock-requested'/'lock-acquired' events"
+                (is (= [:instance-requested :instance-borrowed :instance-returned
+                        :lock-requested :lock-acquired] @events)))))
+          (testing "releasing the lock triggers 'lock-released' event"
+            (is (= [:instance-requested :instance-borrowed :instance-returned
+                    :lock-requested :lock-acquired :lock-released] @events))))))))
```
Comment (KevinCorcoran) [inline test/integration/puppetlabs/services/jruby/jruby_locking_test.clj]: aaaand now I understand why you wrote the File Sync version of this test using an atom.

**Comment by KevinCorcoran** [comment]
I :see_no_evil: 

@camlow325 Do you want to review this or shall I just merge it?

**Comment by camlow325** [comment]
I probably won't get to this today so if y'all want to merge this, I'm good with that.  Otherwise, I might be able to look at it sometime later this week.

**Comment by camlow325** [comment]
:+1:

**Pull request merged**